src/main/java/com/amazon/opendistroforelasticsearch/sql/parser/WhereParser.java

```diff
@@ -522,11 +523,26 @@ private MethodField parseSQLMethodInvokeExprWithFunctionInWhere(SQLMethodInvokeE
         return methodField;
     }
 
+    private MethodField parseSQLCastExprWithFunctionInWhere(SQLCastExpr soExpr) throws SqlParseException {
+        ArrayList<SQLExpr> parameters = new ArrayList<>();
+        parameters.add(soExpr.getExpr());
+        return fieldMaker.makeMethodField(
+                "CAST",
+                parameters,
+                null,
+                null,
+                query != null ? query.getFrom().getAlias() : null,
+                false
+        );
+    }
+
     private SQLMethodInvokeExpr parseSQLBinaryOpExprWhoIsConditionInWhere(SQLBinaryOpExpr soExpr)
             throws SqlParseException {
 
         if (!(soExpr.getLeft() instanceof SQLMethodInvokeExpr
-                || soExpr.getRight() instanceof SQLMethodInvokeExpr)) {
+                || soExpr.getRight() instanceof SQLMethodInvokeExpr)
+                && !(soExpr.getLeft() instanceof SQLCastExpr
+                    || soExpr.getRight() instanceof SQLCastExpr)) {
```

Comment:
Could you refactor this complex condition expression? ex. `bothSidesAreNotFunction() && bothSizeAreNotCast()`